File: benchmarks/rust/hashmap.rs
use std::collections::HashMap;
fn main() {
    let n: i64 = 1_000_000;
    let mut m: HashMap<i64, i64> = HashMap::new();
    for i in 0..n { m.insert(i, i * 2); }
    let mut sum: i64 = 0;
    for j in 0..n { sum += m[&j]; }
    println!("{}", sum);
}

File: benchmarks/cpp/hashmap.cpp
#include <cstdio>
#include <cstdint>
#include <unordered_map>
int main() {
    const int64_t n = 1000000;
    // default-constructed, matching Rust HashMap::new() and Kryos #{} (no pre-size)
    std::unordered_map<int64_t, int64_t> m;
    for (int64_t i = 0; i < n; i++) m[i] = i * 2;
    int64_t sum = 0;
    for (int64_t j = 0; j < n; j++) sum += m[j];
    std::printf("%lld\n", (long long)sum);
    return 0;
}
